### ISSUE TITLE
Wrong indication to upload

### ISSUE THREAD

**Issue by saberraz** [lutraconsulting/input#338]
Input shows my project has been changed as soon as I open them. It should be only marked as changed, if a layer gets changed.

**Comment by saberraz** [comment]
I tested the APK and it works as expected.

**Comment by saberraz** [comment]
Unfortunately this error has resurfaced again.

**Comment by saberraz** [comment]
Seems to  be working from the latest apk on this PR:
https://github.com/lutraconsulting/input/pull/369

**Comment by saberraz** [comment]
OK, this was driving me mad. But finally, I managed to reproduce the bug:
1- I wiped all the storage, cache, app, etc and started from scratch
2- Install Input
3- Log in to mergin
4- Download a project from mergin
5- Open it from Home
6- The project becomes dirty and sync sign will appear

Now, if you close the app after step 4 and continue with opening it, the project does not get dirty upon opening.

**Comment by wonder-sk** [comment]
There's a small binary difference in the gpkg that gets produced. From Viktor's binary diff:

```
< 00000010: 1000 0202 0040 2020 0000 00b1 0000 24e3  .....@  ......$.
---
> 00000010: 1000 0101 0040 2020 0000 00b2 0000 24e3  .....@  ......$.
6,7c6,7
< 00000050: 0000 0000 0000 0000 0000 0000 0000 00b1  ................
< 00000060: 002e 286b 0500 0000 0c0f c000 0000 24e3  ..(k..........$.
---
> 00000050: 0000 0000 0000 0000 0000 0000 0000 00b2  ................
> 00000060: 002e 2c50 0500 0000 0c0f c000 0000 24e3  ..,P..........$.
```

Based on https://sqlite.org/fileformat.html these are the differences in the file header (with offsets):

- 18 - File format write version.   -- 2 vs 1
- 19 - File format read version.    -- 2 vs 1
- 92 - The version-valid-for number. -- 0xb1 vs 0xb2
- 96 - SQLITE_VERSION_NUMBER  -- 0x2e286b vs 0x2e2c50

Needs further investigation what to do with it

**Comment by saberraz** [comment]
The odd thing is, this happens for layers marked as Read-only in the project (in my case, osm.gpkg for background). Maybe, we should have a logic so that the sync will ignore the read-only files.

**Comment by wonder-sk** [comment]
Yeah it happens automatically when opening a .gpkg file. We need to figure out how to best handle that (ignoring read only files is not great in my opinion - that needs to be read from .qgs file, and there is not 1:1 relationship between files and layers - a single .gpkg may contain multiple layers, some read-only, some writable).

**Comment by wonder-sk** [comment]
Also see https://github.com/qgis/QGIS/issues/23991 - essentially the same issue as here

**Comment by wonder-sk** [comment]
@saberraz I believe this should not be a problem anymore in #489
